tests/upgrades/test_capsule.py

```diff
@@ -107,54 +103,56 @@ def test_post_user_scenario_capsule_sync(
+        ak = target_sat.api.ActivationKey(id=pre_upgrade_data.get('ak_id')).read()
+        repo = target_sat.api.Repository(id=pre_upgrade_data.get('repo_id')).read()
+        product = target_sat.api.Product(id=pre_upgrade_data.get('product_id')).read()
+        ak_env = ak.environment.read()
+        org = ak.organization.read()
+        content_view = target_sat.api.ContentView(id=pre_upgrade_data.get('content_view_id')).read()
+        cap_host.nailgun_capsule.content_sync(timeout=3600)
+        cap_host.wait_for_sync(timeout=9000)
+
+        sat_repo_url = target_sat.get_published_repo_url(
+            org=org.label,
+            lce=ak_env.label,
+            cv=content_view.label,
+            prod=product.label,
+            repo=repo.label,
         )
-        ak = target_sat.api.ActivationKey(organization=default_org.id).search(
-            query={'search': f'name={activation_key}'}
-        )[0]
-        repo = target_sat.api.Repository(organization=default_org.id).search(
-            query={'search': f'name={pre_test_name}_repo'}
-        )[0]
-        product = target_sat.api.Product(organization=default_org.id).search(
-            query={'search': f'name={pre_test_name}_prod'}
-        )[0]
-        env_name = ak.environment.read()
-        org_name = env_name.organization.read_json()['label']
-
-        content_view = target_sat.api.ContentView(organization=f'{default_org.id}').search(
-            query={'search': f'name={pre_test_name}_cv'}
-        )[0]
-        capsule = target_sat.api.SmartProxy().search(query={'search': f'name={cap_host}'})[0]
-        call_entity_method_with_timeout(
-            target_sat.api.Capsule(id=capsule.id).content_sync, timeout=3600
+        cap_repo_url = cap_host.get_published_repo_url(
+            org=org.label,
+            lce=ak_env.label,
+            cv=content_view.label,
+            prod=product.label,
+            repo=repo.label,
         )
-        result = execute(
-            lambda: run(
-                f'[ -f /var/lib/pulp/published/yum/http/repos/'
-                f'{org_name}/{env_name.name}/{content_view.name}/custom/{pre_test_name}_prod/'
-                f'{pre_test_name}_repo/Packages/b/{rpm_name} ]; echo $?'
-            ),
-            host=cap_host,
-        )[cap_host]
-        assert result == '0'
+        sat_files = get_repo_files_by_url(sat_repo_url)
+        cap_files = get_repo_files_by_url(cap_repo_url)
+        assert len(sat_files) == len(cap_files) == constants.FAKE_1_YUM_REPOS_COUNT
+        for pkg in constants.FAKE_1_YUM_REPO_RPMS:
+            assert pkg in sat_files, f'{pkg=} is not in the {repo=} on satellite'
+            assert pkg in cap_files, f'{pkg=} is not in the {repo=} on capsule'
+
+        assert sat_files == cap_files
+
+        for file in sat_files:
+            sat_file = target_sat.md5_by_url(f'{sat_repo_url}{file}')
+            cap_file = target_sat.md5_by_url(f'{cap_repo_url}{file}')
+            assert sat_file == cap_file
```

Comment:
Do you think this deep assertion is needed ?

Reply:
Let me answer you the same way I did in https://github.com/SatelliteQE/robottelo/pull/10770/files#r1121370749.

I have just migrated Pulp 2 assertions to Pulp 3 assertions, nothing more. And I have taken those from:
https://github.com/SatelliteQE/robottelo/blob/853f353897f16ef9e8a73cb01ab4470c9a6d7757/tests/foreman/api/test_contentmanagement.py#L1433-L1436

Reply:
Cool!